diagnostic.md

```diff
@@ -5,20 +5,26 @@ Record your responses inside the fenced code blocks below each question.
 1.  Give an example of a visual hierarchy that could be modeled with components.
 
     ```md
-    <!-- your response here -->
+    Nested components
     ```
 
 1.  What is the command to generate a new component called '`my-map`'?
 
     ```sh
-    # your response here
+    ember g component my-map
     ```
 
 1.  What files are edited to produce a component, and what are their
     responsibilities?
 
     ```md
-    <!-- your response here -->
+    * app/components/my-app/component.js
+      * You can define component actions in component.js
+    * app/components/my-app/template.hbs
+      * template.hbs is responsible for the view of the component, and data can be
+      passed around through the template using data-binding.
+    * corresponding template in the outer view (application/template.hbs or app/pokemon/template.hbs from the ember-resources lesson)
+      * where you write the code to display the component
```

Comment:
Looks good.  The `component.js` is also responsible for declaring HTML attributes, and other presentation or behavior, and the `template.hbs` is also responsible for binding and dispatching actions.